rust/sedona-spatial-join/src/exec.rs

```diff
@@ -546,24 +543,21 @@ impl SpatialJoinExec {
 
                     let num_partitions = build_side.output_partitioning().partition_count();
                     let mut build_streams = Vec::with_capacity(num_partitions);
-                    let mut build_metrics = Vec::with_capacity(num_partitions);
                     for k in 0..num_partitions {
                         let stream = build_side.execute(k, Arc::clone(&context))?;
                         build_streams.push(stream);
-                        build_metrics.push(SpatialJoinBuildMetrics::new(k, &self.metrics));
                     }
 
                     let probe_thread_count = self.right.output_partitioning().partition_count();
```

Comment:
```suggestion
                    let probe_thread_count = probe_plan.output_partitioning().partition_count();
```
Isn't this more correct ?

Reply:
I also found that the properties of SpatialJoinExec may be incorrect for KNN join. For instance, [`maintains_input_order`](https://github.com/apache/sedona-db/blob/2939063fbac28dc06a03e1d18743fa327fbba7ed/rust/sedona-spatial-join/src/exec.rs#L215-L223) assumed that the right side is the probe side, which may not be the case for KNN join. We also need to revise the correctness of `compute_properties` of SpatialJoinExec as well.

Reply:
I filed https://github.com/apache/sedona-db/issues/282 to track this issue.